go-controller/pkg/cni/helper_linux.go

```diff
@@ -450,8 +452,64 @@ func ConfigureOVS(ctx context.Context, namespace, podName, hostIfaceName string,
 		ovsArgs = append(ovsArgs, []string{"--", "--if-exists", "remove", "interface", hostIfaceName, "external_ids", types.NADExternalID}...)
 	}
 
-	if out, err := ovsExec(ovsArgs...); err != nil {
-		return fmt.Errorf("failure in plugging pod interface: %v\n  %q", err, out)
+	// 1. plug in ovs port
+	// 2. do following for dpu mode
+	//   2.1 if hw offload is enabled, verify offload by counting tc ingress filter
+	//   2.2 if offload verification fails, retry up to 3 times every 100 ms
+	hwOffloadEnabled, err := isHWOffloadEnabled()
+	if err != nil {
+		return fmt.Errorf("error checking hw-offload flag: %v", err)
+	}
+
+	var link netlink.Link
+	err = retry.OnError(
+		wait.Backoff{
+			Duration: 100 * time.Millisecond,
+			Steps:    3,
+		}, func(e error) bool {
+			if !hwOffloadEnabled {
+				// skip retry if hw offload is not enabled
+				return false
+			}
+			// clean up stale port before retry
+			delPortArgs := []string{
+				"--if-exists", "del-port", "br-int", hostIfaceName,
+			}
+			if out, err := ovsExec(delPortArgs...); err != nil {
```

Comment:
nit: should we ignore `no port named` error as similar to `deletePorts` method ?

Reply:
fixed